File: 2367_Number_of_Arithmetic_Triplets/solution.c
int arithmeticTriplets(int* nums, int numsSize, int diff) {
    int count = 0;
    for (int i = 0; i < numsSize; i++) for (int j = i; j < numsSize; j++) if (nums[j] - nums[i] == diff) for (int k = j; k < numsSize; k++) if (nums[k] - nums[j] == diff) count++;
    return count;
}

File: 2367_Number_of_Arithmetic_Triplets/solution.py
class Solution:
    def arithmeticTriplets(self, nums: List[int], diff: int) -> int:
        n, count = len(nums), 0
        for i in range(n):
            for j in range(i, n):
                if nums[j] - nums[i] == diff:
                    for k in range(j, n):
                        if nums[k] - nums[j] == diff:
                            count += 1
        return count
